Controller/ContactClientDetailsTrait.php

```diff
@@ -95,6 +96,10 @@ public function getEngagements(
             $filters['type'] = $chartFilters['type'];
         }
 
+        if ('revenue' == $filters['type']) {
```

Comment:
Because 'revenue' is actually Stat::TYPE_CONVERTED, having hard checks adds extra noise and complexity, like in StatRepository below...

If it's not possible or too difficult for 'revenue' to be changed to Stat::TYPE_CONVERTED, that's fine...